src/libcore/convert.rs

```diff
@@ -417,6 +434,17 @@ impl<T, U> TryInto<U> for T where U: TryFrom<T>
     }
 }
 
+// Infallible conversions are semantically equivalent to fallible conversions
+// with an uninhabited error type.
+#[unstable(feature = "try_from", issue = "33417")]
+impl<T, U> TryFrom<U> for T where T: From<U> {
```

Comment:
I have no idea if it's better or worse, but could this be on `U:Into<T>`, to pick up more impls?

Reply:
Sounds fine to me. I'd be happy to change it to that if everyone is in agreement to do it.

Reply:
I'd rather not simply because it'd be confusing for `from` to not work but `try_from` to work in those cases.